packages/whook-example/src/repl.ts

```diff
@@ -0,0 +1,50 @@
+import { identity } from '@whook/whook';
+import repl from 'repl';
+import { prepareEnvironment } from './index';
+
+const REPL_BANNER = `
+    _      ____             __     ___  _______  __ 
+    | | /| / / /  ___  ___  / /__  / _ \\/ __/ _ \\/ / 
+    | |/ |/ / _ \\/ _ \\/ _ \\/  '_/ / , _/ _// ___/ /__
+    |__/|__/_//_/\\___/\\___/_/\\_\\ /_/|_/___/_/  /____/
+
+           Inject services with \`.inject\`.
+`;
+
+/* Architecture Note #6: REPL
+
+Here is a simple REPL leveraging the depency injection
+ features in order to let you test things up easily.
+*/
+export async function runREPL(): Promise<void> {
+  try {
+    const $ = await prepareEnvironment();
+    const { $injector, log } = await $.run([
+      ...new Set(['$injector', 'log', 'process']),
```

Comment:
Is process needed ? Cause not used in the object split

Reply:
The `process` service handles the process graceful shutdown. I'm not sure it should stay here but I think that closing connections to DBs properly is always a good idea. That said, it interfers with the REPL `Ctrl+C` handling, leading to the necessity to send the signal 3 times. I'll try to find a better way to handle that.